### PR TITLE
[WIP] added tabs in grands detail page

### PR DESCRIPTION
**Pull request by suhailvs**
##### Description

Added Tabs in grands detail page

##### Checklist

- [ ] Read and conforms to the [contributor guidelines](https://docs.gitcoin.co/mk_contributors/)
- [ ] Read and conforms to the [style guidelines](https://docs.gitcoin.co/mk_styleguide/)
- [ ] Linter status: 100% pass
- [ ] Added relevant tests
- [x] Tests pass and code coverage has not decreased
- [ ] Changes don't break existing behavior
- [ ] Commit message follows [commit guidelines](https://docs.gitcoin.co/mk_contributors/#step-4-commit)
- [ ] [Code documentation](https://docs.gitcoin.co/mk_contributors/#docstrings)
- [ ] Completed: Code review by core team and any requested changes

##### Refers/Fixes
Refs: #3286 

###### Contributor

- [ ] Read and followed the [Contributor Guidelines](https://docs.gitcoin.co/mk_contributors/)
- [ ] Tested all changes **locally**
- [ ] Verified existing functionality
- [ ] Ran `make test` and everything passed!

###### Reviewer

- [ ] Affirm contributor guidelines have been followed and requested changes made
- [ ] CI tests and linting pass
- [ ] No conflicts (migrations, files, etc)
- [ ] Regression tested against staging

###### Funder

- [ ] Validated requested changes were made to specification
- [ ] Bounty payout released to the contributor

### CODE REVIEW

File: app/grants/templates/grants/detail.html [hunk 1]
```diff
@@ -377,4 +412,10 @@ <h5 class="font-body mt-2 font-weight-semibold">
   <script src="{% static "v2/js/grants/shared.js" %}"></script>
   <script src="{% static "v2/js/grants/detail.js" %}"></script>
   <script src="{% static "v2/js/waiting_room_entertainment.js" %}"></script>
+  <script src="{% static "v2/js/pages/tabs.js" %}"></script>
+  <script>
+    $(document).ready(function() {
+      setupTabs('#grant-detail-tabs');
+    });
+  </script>
```
Comment (thelostone-mc) [inline app/grants/templates/grants/detail.html]: Could we shift this into detail.js ? 
Also arrow functions ^_^

Comment (suhailvs) [reply app/grants/templates/grants/detail.html]: ok.

**Comment by suhailvs** [comment]
@thelostone-mc  Will you guide me the following?
I would like to have some sample data to test.

**Task 1: Design the Contributions view**

+ **contributions tab** design

**Task 2: Get details from DB**
how to get the details to fill the contributions tab:
+ date(28 Mar)
+ profile picture
+ name
+ contribution/ month(0.1 ETH/month)
+ txn hash( to create view Etherscan link)
+ transaction amount (0.1 ETH)

here is my try:

    from grants.models import Subscription
    subscriptions = Subscription.objects.filter(grant_id = xx)
    for subscription in subscriptions:
        # profile picture & name
        subscription.contributor_profile.avatar_url
        subscription.contributor_profile.username

**Comment by thelostone-mc** [comment]
@suhailvs 

##### Task1 

the design for the contribution is already present over at https://github.com/gitcoinco/web/blob/master/app/grants/templates/grants/profile/body.html#L28

Just port that code over into a sub template and import it into profile/body.html and the history tab 

##### Task 2 
The BE wiring we'll take care of it later ! 
@captnseagraves is already working on wiring that up.

**Comment by suhailvs** [comment]
@thelostone-mc how did i can create sample grants and do some fund to that grant for testing? when i am trying to create a new grand, the page redirects to typeform.com. 

only if there is some grants and somebody funds that grant, then only i can test.

**Comment by thelostone-mc** [comment]
@suhailvs 
You can either create superuser and assign all permissions to your user / just remove this line

https://github.com/gitcoinco/web/blob/master/app/grants/views.py#L169

**Comment by suhailvs** [comment]
> @suhailvs
> 
> ##### Task1
> the design for the contribution is already present over at https://github.com/gitcoinco/web/blob/master/app/grants/templates/grants/profile/body.html#L28
> 
> Just port that code over into a sub template and import it into profile/body.html and the history tab
> 
> ##### Task 2
> The BE wiring we'll take care of it later !
> @captnseagraves is already working on wiring that up.
> 
> Gimme a ping once you wrap that up and let's get this PR in ^_^

for **Task 1**, do you mean copy the below lines to a separate file say `transactionhistory.html` and import it into `app/grants/templates/grants/profile/body.html` and the new `contributions` tab?

    <h2 class="font-title my-3 pb-3 title">{% trans "Transaction History" %}</h2>
    {% for transaction in history %}
          <div class="py-3 mx-sm-0 row transaction-history">
            <div class="col-3 col-md-1 font-subheader text-center my-auto">
              {{ transaction.cont.created_on|date:"d M" }}
            </div>
            <div class="col-2 col-md-1 justify-content-center special_tag">
              
            </div>
            <div class="col-7 col-md-5 my-auto">
              <div class="title font-body">
                {{ transaction.grant.title }}
              </div>
              <div class="info font-caption">
                {{ transaction.sub.amount_per_period }} {{ transaction.sub.token_symbol }} / {{ transaction.sub.frequency }} {{ transaction.sub.frequency_unit }}
              </div>
            </div>
            <div class="col-5 col-md-2 font-body my-auto txn-link">
              <a id="tx_link" href="" target="_blank" rel="noopener noreferrer">
                View Etherscan
              </a>
              <input type="hidden" id="tx_id" name="tx_id" value="{{ transaction.cont.tx_id }}"/>
            </div>
            <div class="offset-lg-1 col-7 col-md-2 my-auto tags font-caption">
              <div class="tag tag-lg token">
                <p>
                  {{ transaction.sub.amount_per_period }}
                  <span>{{ transaction.sub.token_symbol }}</span>
                </p>
              </div>
              {% if transaction.value_in_usdt_now %}
                <div class="tag tag-lg usd">
                  <p>
                    {{ transaction.value_in_usdt_now }}
                    <span>USD</span>
                  </p>
                </div>
              {% endif %}
            </div>
          </div>
    {% endfor %}

**Task 2** will be taken care by you?

**Comment by owocki** [comment]
It's trivial to add some Grants / Subscriptions / Contributions via the Django admin.  I think you should be able to do it @suhailvs if you spend some time with Django.

> for Task 1, do you mean copy the below lines to a separate file say transactionhistory.html and import it into app/grants/templates/grants/profile/body.html and the new contributions tab?

This sounds  directionally correct to me.

From the original Issue

> Pls only start work on this PR if you can submit a WIP PR in the next 3 days.

@suhailvs when do you think you can have a feature complete PR? I did not expect that this would drag on for several days.  I appreciate your interest, but this is an important functionality for us -- If you cannot get it wired up, perhaps we should give you another bounty that matches ur skillset?

**Comment by suhailvs** [comment]
@owocki sorry. i am out.

**Comment by owocki** [comment]
thanks for letting us know!

**Comment by thelostone-mc** [comment]
Closing this out !

**Pull request closed without merging**